### DISCUSSION TITLE
accessing the md_ast to html function

### DISCUSSION THREAD

**Discussion by rambip** [wooorm/markdown-rs#63] [Q&A]
Hi !

I'm using your awesome work to create a web-app to display markdown notes, but I'm stuck.
I call the [katex](https://katex.org/) javascript library from rust to translate the latex to a svg format.

I have done the change directly on the mst (see [here](https://pastebin.mozilla.org/wHWDrvzo) ), but I don't know how to convert it back to html

Am I missing something ?
Is there a simpler way to do what I want ?

Thanks in advance

**Comment by wooorm** [comment]
There is no current utility that turns mdast into a string of html

**Comment by rambip** [comment]
Oh ok I was convinced the to_html function use the abstract syntax tree !

In that case, I think that would be important to add a way to provide your own function that maps latex content to html

Another issue I had was that $<$ was translated to `&lt` in the html so it made the latex rendering fail afterward ...

**Reply by wooorm** [reply]
> In that case, I think that would be important to add a way to provide your own function that maps latex content to html

Yes, a feature like that might be useful

You can also run KaTeX in a browser.

> Another issue I had was that  was translated to `&lt` in the html so it made the latex rendering fail afterward ...

Your input and output is messed up. Can you please use code to show your actual input and the actual output? Also, this should not affect KaTeX.

**Reply by rambip** [reply]
> You can also run KaTeX in a browser.
> 

I think I didn't explain my project well enough: I use yew, the rust web framework, to convert the markdown to html in the browser. So I use katex in the browser ! (I just call it through my rust code). That my explain why the < is translated to `&lt` at some point, when I try to load the content in a html element.

After thinking about it I think I will implement my own methods to display the ast as html: it will be easier to maintain with yew and allow more flexibility.
Yew provide a convenient `html!` macro, so it will not be to bad to implement
